sonar-kotlin-plugin/src/main/java/org/sonarsource/kotlin/api/CollectionExtensions.kt

```diff
@@ -0,0 +1,37 @@
+/*
+ * SonarSource Kotlin
+ * Copyright (C) 2018-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.kotlin.api
+
+/**
+ * Returns `true` if the predicate does match for all element pairs (`this[i]`,`other[i]`) of both lists.
+ * This implies that both lists must have the same size.
+ */
+@Generated
+inline fun <T, U> List<T>.allPaired(other: List<U>, predicate: (T, U) -> Boolean) =
+    size == other.size && run {
+        forEachIndexed { index, value ->
+            if (!predicate(value, other[index])) {
+                return false
+            }
+        }
+        true
+    }
+
+private annotation class Generated
```

Comment:
for now, it's okay to keep it like this. However for the future, I'd find out a better solution for coverage